src/widget/wcoverartmenu.cpp

```diff
@@ -82,8 +88,28 @@ void WCoverArtMenu::slotChange() {
     coverInfo.source = CoverInfo::USER_SELECTED;
     coverInfo.coverLocation = selectedCoverPath;
     coverInfo.setImage(image);
-    qDebug() << "WCoverArtMenu::slotChange emit" << coverInfo;
-    emit coverInfoSelected(coverInfo);
+
+    QString coverArtCopyFilePath =
+            trackFileInfo.absoluteFilePath().left(
+                    trackFileInfo.absoluteFilePath().lastIndexOf('.') + 1) +
+            selectedCoverExtension;
+
+    if (QFileInfo(m_coverInfo.trackLocation).canonicalPath() ==
+            QFileInfo(selectedCoverPath).canonicalPath()) {
+        qDebug() << "Track and selected cover art are in the same path:"
+                 << QFileInfo(selectedCoverPath).canonicalPath()
+                 << "Cover art updated without copying";
+        emit coverInfoSelected(coverInfo);
+        qDebug() << "WCoverArtMenu::slotChange emit" << coverInfo;
+        return;
+    }
+
+    m_worker.reset(new CoverArtCopyWorker(image, coverArtCopyFilePath));
+    m_worker->run();
+    if (m_worker->isCoverUpdated()) {
```

Comment:
This is pointless, because we need to wait for the worker thread to be updated.

Reply:
So we need to have signal&slots right?

Reply:
Do we need a feedback in the GUI? If yes, or does it already work via the track object signals? 
If we need a feedback, we need a signal slot connection.

Reply:
I guess we don't need a feedback in the gui, because cover art updated after the thread ended (atleast for now, there is a TODO to move that in another thread as we discussed). I thought we could do it with the signal&slots, I have searched about it a bit now, can this be achieved by using [QFuture::waitForFinished()](https://doc.qt.io/qt-5/qfuture.html#waitForFinished)?

Reply:
Oh, this slipped through. I think this condition can be removed.

Reply:
It needs to be replaced with a signal slot connection from the worker thread.

Reply:
Added signal&slot 👍